### ISSUE TITLE
Refuelling stops immediately 

### ISSUE THREAD

**Issue by KBtwentytwenty** [Fragtality/Fenix2GSX#11]
I downloaded this last night, it’s really cool! Thanks for your work. 

I’ve come across an issue (maybe user error) where the refuelling stops almost immediately after starting. 
I have followed the readme exactly and tried with the advised options disabled and enabled. 
If I have 3000kg in the tank and the required is 9000kg, for example, the fuel truck connects, loads 28kg, (or whatever I set the number at) and then ceases the fuelling and drives off.

As I mentioned above, thanks for this, would just be really nice to be able to have fuelling work, so any help is appreciated.
Cheers!

**Comment by Fragtality** [comment]
Hmm, that is strange! It should load the Amount planned by SimBrief.
Is the SB Flightplan also in kg?

The Log-File from the Time where it happened would maybe be useful (Fenix2GSX<date>.log).

**Comment by KBtwentytwenty** [comment]
[Fenix2GSX20230523.log](https://github.com/Fragtality/Fenix2GSX/files/11547606/Fenix2GSX20230523.log)

The SB fp is also in kg, all the settings are according to the readme. I've tried with different fp's and after restarting, etc. Log says waiting for next truck, but that never comes.
Thanks for the fast response.

**Comment by Fragtality** [comment]
Does it work on a Parking Position with Underground Fuel?

Does the Fuel on the Plane actually change (on the ECAM)? Maybe the Binary is blocked by some kind of Security/Firewall Software.

**Comment by KBtwentytwenty** [comment]
I have only tried at an underground fuel. It refuels by on the efb, and on the ecam, but with 2 different values.

![image](https://github.com/Fragtality/Fenix2GSX/assets/66142366/c42b2abd-87fe-48bd-a5fa-f15b0dbf910b)
![image](https://github.com/Fragtality/Fenix2GSX/assets/66142366/d815eafe-e915-42b0-a88c-120921728d0c)

**Comment by KBtwentytwenty** [comment]
To add to my previous comment, it seems the fuel on the efb decreases, and the ecam doesn't, whilst still showing the refueling message.

**Comment by Fragtality** [comment]
That's normal for the ECAM, it does not show the last digit and does a bit of rounding.

What I find strange is the Cargo - it is nearly twice as much as planned and Cargo is already loaded before the Passengers. Seems like something (or someone?) else is manipulating the Mass and Balance of the Plane!?


> To add to my previous comment, it seems the fuel on the efb decreases, and the ecam doesn't, whilst still showing the refueling message.

That does not make Sense at all. The Fuel Amount will always progressively increase. The only time it is reduced is when your current FOB is over the planned Amount, but that happens instantly (and already before the Truck arrives)

**Comment by Fragtality** [comment]
Please send me Screenshots for the GSX Settings and Aircraft Profile:
- GSX In-Game Menu -> GSX Settings -> Timings
- GSX In-Game Menu -> GSX Settings -> Simulation
- GSX In-Game Menu -> Customize Aircraft

**Comment by Fragtality** [comment]
Hmm, looking normal on first Glance.

What other Addons are you using / running? (Functional, so not Sceneries and Airplanes)

**Comment by KBtwentytwenty** [comment]
Only thing i could think of, is does your program select the incorrect fuel option from the menu, ie not the one with the simbrief amount?

**Comment by Fragtality** [comment]
It takes the planned Amount as known by the Fenix.
Which Menu?!

Is Fs2Crew running?

**Comment by KBtwentytwenty** [comment]
Sorry, should have been more specific. I thought your program selected an option from the gsx menu, but as you say its the fenix planned fuel, so that cant be the case.
And no, fs2crew isnt running and hasnt been running since i installed your addon

**Comment by KBtwentytwenty** [comment]
Been trying to troubleshoot a bit more since yesterday.
Reinstalled the app - same problem occurs
Pressed request refueling manually in gsx menu after the app sent for automatically - same problem occurs

In the logs, it says 'waiting for next truck' but that makes no sense as it's an underground fueling spot

Every time i manually press 'request refuelling', the amount in tanks increases by 20kg.
Boarding does not automatically start as I assume the program knows the refueling hasnt completed?
ECAM shows refueling indefinitely, suggesting the program knows it hasnt finished?

Hopefully this helps you understand what my problem could be. I definitely think it isn't down to user error.

**Comment by KBtwentytwenty** [comment]
Oh and left it running a bit longer and the fuel is taken out of the tanks! Shown on both ECAM and EFB

**Comment by Fragtality** [comment]
> Hopefully this helps you understand what my problem could be. I definitely think it isn't down to user error.

1) No, since repeating the same Steps or calling an already called Service does not make any Difference to the Situation and it is clear that it yields the same Results. That kind of Troubleshooting does not give any Hints, because you did not really try/do something in a different Way.
2) No, since I already know the Cause: GSX is led to believe it can already disconnect the Fuel Hose. The Root-Cause for that is unclear.
3): Well, we can call it Setup-Specific-Error if that makes it more comfortable. But if it would be a general or common Bug it would have been already known and either fixed or documented by now.

<br/>

> Oh and left it running a bit longer and the fuel is taken out of the tanks! Shown on both ECAM and EFB

APU Running? Does it run every Time you used the Tool? What is your Default-State for the Plane?

**Comment by KBtwentytwenty** [comment]
Sorry, didn't mean for the user error part to sound ignorant, I just can't see what I could have done wrong!

Default is cold and dark and then I switch on the plane before syncing flight plan. I now realise my dumb self forgot the apu needs fuel (duh) so thats that 'problem' fixed.

Are there any steps I could do to help with troubleshooting? If not I'll just deal with it and hope that it fixes itself at some point or I find a fix.
Thanks very much for your help and the addon!

**Comment by Fragtality** [comment]
Nothing to be sorry, I did not perceive that as ignorant - it should just express that we can not exclude anything at the Moment ;)

When you switch on the Plane - what exactly happens there (until you load the Flightplan in the EFB)?

I'm currently a bit out of Ideas to be honest - can't really grasp what might be the Trigger/Cause for GSX to disconnect the Hose so fast. That makes it hard to formulate any Troubleshooting Steps ^^

**Comment by KBtwentytwenty** [comment]
This is the process ->
1. Load up MSFS, open Fenix2GSX, GSX opens along with the game
2. Select Gate and load into world, press ready to fly
3. Fenix2GSX connects, after 3 seconds repositioned. Jetway connets. Aircraft is cold and dark
4. Ground Power on, batteries on, aircraft systems startup, standard procedure etc. 
5. Go to flypad, take from simbrief, underground fuel truck and catering arrive, connect.
6. Fuel truck starts, runs for about 10 seconds or a couple dozen kilograms, disconnects.
7. Catering vehicles and fuel truck drive off, boarding doesn't begin

**Comment by Fragtality** [comment]
What is "standard procedure etc." in Step 4?

I'm not so sure anymore that the Cause is GSX removing the Hose to early. Maybe it is correct since the FOB is not increasing anymore (also the Trigger during normal Operation of the Tool).
But then I'm equally clueless at the Moment why the Binary would only increase the FOB one Time and then does not continue 🤔

**Comment by KBtwentytwenty** [comment]
GPU, Batteries, ADIRS, Lights, APU, and then i leave the plane alone whilst i wait for the fuel truck

**Comment by Fragtality** [comment]
![image](https://github.com/Fragtality/Fenix2GSX/assets/78614902/6a19cc38-45e0-4987-af00-d56352edd699)
APU?!

Have you tried without starting the APU before Refueling? (Is not really necessary either ... your Power comes from the Ground and AC is provided through PCA)

**Comment by Fragtality** [comment]
That's why I don't like YT Tutorials 😅 
Though it is not entirely wrong. Most (european? bigger?) Airports prohibit / restrict the Usage of the APU, if there are other means for Power and AC. Company-SOPs on top of that also differ widely. But let's say you are on a remote Stand on a hot Summer Day ... then you probably want to and are allowed to start it for cooling the Cabin (if there is no Cart for PCA available at that Airport of course). You see: there is just no flat general Rule for that ;)
That is why the default Settings of Fenix2GSX are like they are regarding PCA: It will be connected on Jetways, but on Stand without Jetway it will not be connected. Should be most of the Time right ^^

Damn - I somehow hopped that could be it :/ 

When you say "Lights" - that still means the Beacon is off, right?

**Comment by KBtwentytwenty** [comment]
Yeah man, I watched one tutorial when first learning the fbw, and stuck with the procedure ever since!

And yeah, only lights turned on are nav/logo and interior. 
Its really odd because Fenix2GSX knows it hasn't yet finished refueling, but gsx just doesnt continue.
the log just ends here
![image](https://github.com/Fragtality/Fenix2GSX/assets/66142366/c156ba41-e200-499f-ab82-204747a46ada)

**Comment by Fragtality** [comment]
It is either GSX removing the Hose too early, or it is Fenix2GSX not continuing to increase the Fuel Amount (which would also trigger GSX).
GSX does not know what the correct Fuel Amount is - it wait until the Fuel Amount stopped increasing. That's why a bit of Refuel-Synchronization could also be achieved before Fenix2GSX.

Which makes we wonder if that would work in your Setup ... ?!
Do that Test without Fenix2GSX running at all:
- Power up procedure, load the FP in the EFB (but keep the APU off in that Case to be sure)
- Call GSX Refueling and wait until the Hose is connected to the Plane (visually)
- Start Fast or Realtime Loading in the EFB (not instant!)

Does the Fuel-Track also leave that early or does it wait until the FOB has reached the planned Amount?

**Comment by KBtwentytwenty** [comment]
Using realtime loading, the fuel truck doesnt leave until the planned FOB amount is loaded.
No changes in procedure compared to when using Fenix2GSX.

**Comment by Fragtality** [comment]
Hmm 🤔

**Comment by KBtwentytwenty** [comment]
![image](https://github.com/Fragtality/Fenix2GSX/assets/66142366/6a203991-1716-4a99-b3b6-90dd6fb98834)
filled all the way to planned, whereas using Fenix2GSX it would stop at around 3028kg and disconnect

**Comment by FGrabi** [comment]
Hello,
I'm facing exactly the same problems and symptoms. Only difference, I'm using fs2crew

**Comment by Fragtality** [comment]
Well, KBtwentytwenty also has fs2crew installed (but not opened). So it is one thing both of your Setups have in common 🤔 

Have you also tried if the "Fuel-Synchronization" works without Fenix2GSX (see above for the Steps)? Would be interesting to know if that Behavior is the same in your Setup!

Also for you both: Please test if the Boarding-Integration would work. So start a Session with Fenix2GSX, but disable the Option that Refueling is called automatically before that. When you have imported the Flightplan in the EFB, call Boarding from the GSX Menu. Does that work?

But I'm still a bit clueless to be honest, I think I need to prepare a Version with more Logging-Outputs in the Hope that this would give some kind of Idea :/

**Comment by KBtwentytwenty** [comment]
Hi. Thanks for your addition FGrabi.

Just disabled Fenix2GSX automatic fuelling and manually called for boarding. Payload and PAX count sync as expected.
This shows that its only a problem with refuelling, right?

**Comment by Fragtality** [comment]
Yeah, would be my Assumption too - so there is no general Error in the Communication with the Fenix at least!

**Comment by KBtwentytwenty** [comment]
Hopefully that narrows it down slightly for you! I am no dev whatsoever, I wish I could help more as this has really added to my flightsim experience in the short time I've been using it.
Thanks for what you're doing :)

**Comment by FGrabi** [comment]
Hi,
tested the following:
1. Start MSFS without FENIX2GSX
2. Placed Fenix C&D at Gate in KLAX (inibuilds Payware)
3. Reposition Aircraft at Gate via GSX Menu
4. Operate Jetway (GSX)
5. Power on EFB
6. SIM Settings in EFB (Standby:ISIS, Linked Baro, Brake FAN - Checked that weight in kg and distance metres)
7. Start FS2CREW Fenix and RAAS - set callsign in FS2CREW
8. Power Up Aircraft via FS2Crew (Battery, Firetest, ADIRS, Ground Pwr, NO-APU, Lights (NAV-Strobe/Auto,)
9. Loading Simbrief Flightplan in EFB
10. GSX - Request Refuel
11. After a few seconds: GSX Request Catering
12. Loading FP in MCDU and INIT-A,
13. When Refuel Truck is connected (checked that hose is visible copnnected, worker stood still): Press Load Aircraft - Realtime)
14. Truck starts refuel until FOB in one-go (Tank was 3.000, After refuel 11.822) 

Will test the FENIX2GSX Boarding Integration without automatic refuel next

**Comment by FGrabi** [comment]
Test with Boarding:
1. Start MSFS and Fenix2GSX (Autostart via MSFS-Addon-Linker)
2. same as above
3. Done by Fenix2GSX
4. Done by Fenix2GSX
5. to 9. same as above
10. GSX - Request Boarding
11. When Boarding Sound starts in GSX and GSX PAX Counter increases - press Load Aircraft Realtime in EFB
12. Pax, cargo and fuel increase in EFB Pax out of sync with GSX - for sure, no fueltruck visible
13. GSX finished boarding, cargo in EFB shows correct planned kg but PAX in EFB showing only 54 on board from 160 planed and loaded by GSX/Fenix2GSX
14. After Boarding the refuel truck was requested by Fenix2GSX - the fuel in EFB was still "loading" missing ~2.000 kg
15. Truck connected and automatically starts refuel (visible at the truck and by GSX Message).
16. When planed fuel reached, truck disconnects and drive away
17. PAX in EFB still "boarding"
18. Stoped MSFS

**Comment by Fragtality** [comment]
Thanks! So the "manual Refuel Synchronization" works for you too.

To be brutally honest: the second Test is useless. It is either Loading via EFB or using the GSX-Integration. Using both is out of Scope. Please do it like described: Disable Auto-Refuel, Start Session, Load Flightplan in EFB, Request Boarding via GSX.

Would it be possibly for one of you to completely uninstall FS2Crew Fenix (RAAS is fine, I have that myself)? Maybe it still somehow active even when not opened?

**Comment by Fragtality** [comment]
Had a Chat with Fs2Crew: There might be something that could remotely possible have an Influence.
There should be some kind of "Fuel Truck" Checkbox in your Fs2Crew Settings. Please uncheck that and test in a new Session after that!
If that also does not help, please try the latest experimental Version of both Fenix Edition and RAAS (so that their WASM for L-Var Communications is not needed anymore).

Does any of these Recommendations make a Difference?

**Comment by FGrabi** [comment]
Installed FS2Crew Fenix experimental version. Reactivated Automatic Refuel after FP import. Start MSFS, Fenix2Crew Loop starts (reposition, jetway) waiting on FP import. Power up Aircraft with fs2crew. Import FP in EFB. Automatic Refuel request from Fenix2GSX - followed by Catering. Truck arrives, connected to the Tank and Aircraft. Fills in 86 kg (Initial Level 3000kg, planned 11272kg) Fuel truck disconnects and go away. Same message - waiting for next truck. Shortly after GSX message "Refuel finished". Calling a new truckl with GSX: Same result.
Next test: I removed FS2Crew Fenix via FS2Crew Installer. Will disable FS2Crew Command Center during MSFS startup (via MSFS-Addon Linker, because EXE.XML does not work with programms that need admin rights) Will report results

**Comment by FGrabi** [comment]
Without FS2Crew (Deinstalled FENIX Version, no running Command Center) same problems. So it seems, that it is not an fs2crew problem
Will try to test a little bit more with the GSX profile for the airport

**Comment by Fragtality** [comment]
Thanks for your Tests!

I don't think that GSX Profiles could cause that. They more or less only define which 3D Object and Animation is used for the Truck. The Mechanics are the same.

Admin Rights, huh? So what are you both starting with Administrator Rights and what with normal Rights?
MSFS, GSX, Fenix, Fenix2GSX and FS2Crew (I only have RAAS) all run on the same normal User-Elevation for me without any Problem. And it would be not the first Strange Issue that would be caused by not running everything on the same Elevation (the first for Fenix2GSX, but not the first across my Tools).

**Comment by FGrabi** [comment]
Running most of the Applications **without** Admin rights:

- MSFS
- GSX
- Fenix2GSX

**With** Admin Rights
- FS2Crew (needs installed with Admin Rights - see Install instructions for the experimental versions)
- MSFS AddonLinker (because some of the Payware airports need also admin rights during install, e.g. inibuilds
- Navigraph Navdata is installed with admin rights

**Comment by Terminal-Rat** [comment]
I'm also having this same issue. No FS2Crew, doing the manual fuel synchronization with GSX without Fenix2GSX works just fine. Nothing is set to run elevated.

**Comment by Terminal-Rat** [comment]
> Are any of you running any AI-Injecting Tools? FSLTL / FS Traffic / AIG / Default AI?

I have AIG models but I don't have it injecting anything. I also just tried with an empty community folder except for the fenix, GSX, and the mobiflight WASM module and it disconnected early again.

**Comment by KBtwentytwenty** [comment]
> Are any of you running any AI-Injecting Tools? FSLTL / FS Traffic / AIG / Default AI?

All of the above, but not started

**Comment by FGrabi** [comment]
Running active in paralell to Fenix2GSX and couatl

**Startet via MSFS AddonLinker**
- NavigraphSimlink
- AIGround.exe (Freeware from flightsim.to - addon for FSTLT and AIG))
- AIFlow.exe (Freeware from flightsim.to - addon for FSTLT and AIG)
- PilotCore (connector for IVAO)
- FSTLT Injector
- FBW Simbridge

**Startet via EXE.xml**
- vdgs-util.exe
- aerosoft-vdgs-driver
- FENIX Bootstrapper
- FSUIPC7.exe (free version)
- FS2Crew Command Center

**Comment by Fragtality** [comment]
So no real Commonality there :/

@Terminal-Rat  @KBtwentytwenty : What about the AeroSoft VDGS Driver and FBW SimBridge? Do you run these too?

@FGrabi Have you tried without FSTLT and the AI-Tools?

**Comment by KBtwentytwenty** [comment]
I haven't got the VDGS driver installed, nor do I run SimBridge when using the Fenix. But that is installed

**Comment by Fragtality** [comment]
@KBtwentytwenty @FGrabi @Terminal-Rat 
Please download [RC2](https://github.com/Fragtality/Fenix2GSX/releases/tag/v0.3.1-RC2) and enable Verbose Logging: In the File Fenix2GSX.dll.config, the key logLevel from Debug to Verbose. Please do me a Favor and either deactivate VHF1 Audio Control or make sure your VHF1 is running and has an Audio-Session open. (Else the Logs gets harder to read than neccessary)
With that, do a "normal" Sim-Session. Wait some Time after the Refuel stopped and end the Session. Send me the Log of that Session.

**Comment by KBtwentytwenty** [comment]
Downloaded now, will report back soon

**Comment by KBtwentytwenty** [comment]
Ok so this is odd. I disabled VHF1 and now it seems to be fuelling all the way, Hasn't finished yet and I'll still send the log here, but aside from that, no changes to starting procedure

**Comment by Fragtality** [comment]
Okay, that is strange oO
And with VHF1 enabled?!

**Comment by Terminal-Rat** [comment]
Here's my log, of course it didn't do it this time... Only difference between this and previous times is that I'm using RC2 and I have the two audio options deselected. I also tried with the GSX audio control option selected, with that and VHF1 selected with a Vpilot session, and with that and VHF1 selected without a vpilot session and it didn't happen any of those times either. 
[Fenix2GSX RC2 test 1.txt](https://github.com/Fragtality/Fenix2GSX/files/11638626/Fenix2GSX.RC2.test.1.txt)

**Comment by KBtwentytwenty** [comment]
> Okay, that is strange oO And with VHF1 enabled?!

with VHF1 Enabled, the issue still occurs

[Fenix2GSX20230602.log](https://github.com/Fragtality/Fenix2GSX/files/11638758/Fenix2GSX20230602.log)

**Comment by KBtwentytwenty** [comment]
I still had the GSX audio option enabled, only change being to VHF1

**Comment by Fragtality** [comment]
Why do you say the Issue is still occuring, I can see in the Logs that the Refuel was working?

**Comment by KBtwentytwenty** [comment]
The issue occurs when VHF1 is enabled, as you asked here

> Okay, that is strange oO And with VHF1 enabled?!

**Comment by Fragtality** [comment]
Then please the Log when the Issue (Refuel stops immediately) occurs :)

**Comment by KBtwentytwenty** [comment]
[Fenix2GSX20230602.log](https://github.com/Fragtality/Fenix2GSX/files/11638885/Fenix2GSX20230602.log)
with VHF1 enabled

**Comment by Fragtality** [comment]
I think I've found it ... there are so many Audio-Devices with so many Audio-Sessions that the Search for the VHF1 App takes so long that the Tool can't increase the Fuel in time.

**Comment by KBtwentytwenty** [comment]
Lol, that is so annoying!
Such a simple answer and all caused because of the billion audio outputs I have.
Now that I know the issue is with VHF1, I'll just keep it disabled. The addon works perfectly other than that, thanks for your help.

**Comment by Fragtality** [comment]
@KBtwentytwenty @FGrabi @Terminal-Rat 
Please try [RC3](https://github.com/Fragtality/Fenix2GSX/releases/tag/v0.3.1-RC3)
Does that work with Audio Control enabled but the VHF1 App not running?

**Comment by Fragtality** [comment]
Nice!

How does it work out for the other two? :)

**Comment by FGrabi** [comment]
Hi,

sorry for the delay, was busy with my "real" job ;-)

Checked the RC3 version together with all my Addons. The RC3 version works perfect. Refuel and catering start after loading FP, Refuels the planned amount of Fuel, After 90 sec, boarding starts and is synced with the Fenix EFB, PAX boarding finished before cargo, but when the cargo was finished (GSX) the loaded Cargo in the EFB is still ~400kg less than the planned cargo. I believe This is most likely  a "feature" of the Fenix Simbrief Integration. Will request support in the Fenix Discord.

I like to thank you for the support and for this excellent application!! Love it!

Question - enhancement request: Will it be possible - maybe in one of the future releases - to set the wait times for boarding and final loadsheet or include them in a loop depending on variables/triggers? Something like:

- Start boarding if refuel and catering is finished (message from GSX) **and** No-Smoking sign is switch to ON (after refuel) **_or_** after xx secons (adjustable config parameter)
- Send final loadsheet if boarding (and refuel/catering) finished (GSX message) **and** APU bleed is on _**or**_ after xx seconds (adjustable config parameter)

Thanks again
Frank

**Comment by Fragtality** [comment]
> Checked the RC3 version together with all my Addons. The RC3 version works perfect.

Well, due to the nature of the Bug, it would also be great to test without the VHF1 App running (but the Option being enabled). 😉 

<br/>

>  but when the cargo was finished (GSX) the loaded Cargo in the EFB is still ~400kg less than the planned cargo.

So it is 400kg less than what it shown in the EFB as planned (Mass & Balance)? NOT what is planned in SimBrief?
I'll keep an Eye on it, never saw that Behavior before. 🤔 

<br/>

> I like to thank you for the support and for this excellent application!! Love it!

Thanks 😃 

<br/>

> Question - enhancement request: Will it be possible - maybe in one of the future releases - to set the wait times for boarding and final loadsheet or include them in a loop depending on variables/triggers?

yes-no.
Sure that is doable. But I come to a point where I generally won't add/change any Automation Feature. Since everyone wants their personal Flow/Sequence of Services and the Tool is already at a Point, where it is debatable overloaded. So adding more and more Options to make every possible Flow configurable does not sound reasonable to me.
But: since the Load-Integration/Synchronization of Fenix2GSX is always active, everyone has the Option do write their own Automation. (For example with FSUIPC in a Lua-Script)

> Start boarding if refuel and catering is finished (message from GSX) and No-Smoking sign is switch to ON (after refuel) or after xx secons (adjustable config parameter)

That is only a slight nuance, since that is already the Case mostly. Sure, Fenix2GSX does not look for the Signs, but Boarding is started 90s after Refueling & Catering (if configured) are finished. Lower does not make sense, since the Fuel-Truck wont have left. A longer Delay also does not make that much sense, since the 90s are already too much for some (or even waiting for the Refuel to complete).
Adding another Constraint to the Boarding-Start would not really improve the Ease-of-Use. So that would require an Option again.

> Send final loadsheet if boarding (and refuel/catering) finished (GSX message) and APU bleed is on or after xx seconds (adjustable config parameter)

I don't really see the Benefit of adding another Constraint to the Final LS, especially to the APU on top of that. When Boarding/Refueling is finished (whoever called the Service), it will be send after a random Delay (90-150s). It even was made more independent in the recent Versions to allow different SOPs to be followed. So that again would require yet another Option.

**Comment by KBtwentytwenty** [comment]
Thanks, all is good. Appreciate your help and work
